File: template_backend/src/main/java/com/benny1611/template/dto/PasswordResetConfirmRequest.java
package com.benny1611.template.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.Pattern;
import lombok.Data;

import java.util.UUID;

@Data
public class PasswordResetConfirmRequest {
    @NotBlank
    private String secret;

    @NotNull
    private UUID tokenId;

    @NotBlank(message = "password cannot be empty")
    @Pattern(
            regexp = "^(?=.*[A-Za-z])(?=.*\\d)(?=.*[@$!%*#?&])[A-Za-z\\d@$!%*#?&]{8,}$",
            message = "Password must be at least 8 characters long and contain a letter, a number, and a special character"
    )
    private String newPassword;
}

File: template_frontend/easyevent/src/models/dto/PasswordResetConfirmRequest.ts
export class PasswordResetConfirmRequest {
    secret: string;
    tokenId: string;
    newPassword: string;

    constructor(secret: string, tokenId: string, newPassword: string) {
        this.secret = secret;
        this.tokenId = tokenId;
        this.newPassword = newPassword;
    }
}

export default PasswordResetConfirmRequest;